### PR TITLE
Changes for 0.0.8

### PR DESCRIPTION
**Pull request by Zhyrek**
* Fixed circular dependency between field and simulation, field imports simulation only when you create a new field object, in order to set a default Simulation object to assign the field to
* Changed field to use cell spacing of parent Simulation object

### CODE REVIEW

**Comment by Zhyrek** [comment]
Also removed pycalphad as mandatory dependency

File: pyphasefield/pyphasefield/field.py [hunk 1]
```diff
@@ -24,41 +28,49 @@ def __array_finalize__(self, obj):
...
-        result = (np.roll(self, -1, 0) + np.roll(self, 1, 0) - 2*self)/(self._cell_spacing**2)
-        for i in range(1, self._number_of_dimensions):
-            result += (np.roll(self, -1, i) + np.roll(self, 1, i) - 2*self)/(self._cell_spacing**2)
+        number_of_dimensions = len(self.shape)
+        inverse_cell_spacing = 1./(self._simulation._cell_spacing_in_cm)
+        inverse_dx_squared = inverse_cell_spacing**2
+        result = (np.roll(self, -1, 0) + np.roll(self, 1, 0) - 2*self)*inverse_dx_squared
+        for i in range(1, number_of_dimensions):
+            result += (np.roll(self, -1, i) + np.roll(self, 1, i) - 2*self)*inverse_dx_squared
         return result
```
Comment (lucentdan) [inline pyphasefield/pyphasefield/field.py]: Is this something that the user has to choose when developing an engine?

File: pyphasefield/pyphasefield/field.py [hunk 2]
```diff
@@ -24,41 +28,49 @@ def __array_finalize__(self, obj):
...
+        inverse_cell_spacing = 1./(self._simulation._cell_spacing_in_cm)
+        array = (np.roll(self, -1, 0) - self)*inverse_cell_spacing
         array = array[np.newaxis] #add dimension for different partial derivatives
-        for i in range(1, self._number_of_dimensions):
-            temp = (np.roll(self, -1, i) - self)/(self._cell_spacing)
+        for i in range(1, number_of_dimensions):
+            temp = (np.roll(self, -1, i) - self)*inverse_cell_spacing
             temp = temp[np.newaxis]
             array = np.concatenate((array, temp), axis=0)
         return array
```
Comment (lucentdan) [inline pyphasefield/pyphasefield/field.py:65]: Is this something that the user has to choose when developing an engine?

File: pyphasefield/pyphasefield/field.py [hunk 3]
```diff
@@ -24,41 +28,49 @@ def __array_finalize__(self, obj):
...
+        inverse_2dx = 0.5*inverse_cell_spacing
+        array = (np.roll(self, -1, 0) - np.roll(self, 1, 0))*inverse_2dx
         array = array[np.newaxis] #add dimension for different partial derivatives
-        for i in range(1, self._number_of_dimensions):
-            temp = (np.roll(self, -1, i) - np.roll(self, 1, i))/(2*self._cell_spacing)
+        for i in range(1, number_of_dimensions):
+            temp = (np.roll(self, -1, i) - np.roll(self, 1, i))*inverse_2dx
             temp = temp[np.newaxis]
             array = np.concatenate((array, temp), axis=0)
         return array
```
Comment (lucentdan) [inline pyphasefield/pyphasefield/field.py:43]: This is a centered gradient...is this something that will need to be known by the engine?  Or will all that happen through the names of the functions?

**Review by lucentdan — APPROVED** [review]
I agree with these changes and would like to see them in action.

File: pyphasefield/pyphasefield/field.py (hunk 2 above)
Comment (Zhyrek) [reply pyphasefield/pyphasefield/field.py:65]: Yes, both face_gradient functions essentially do the same thing, but are mirrored. Both return an array one dimension larger than the original, with each value along this dimension representing which piece of the gradient vector it corresponds to. face_gradient_right computes the face gradient using the central cell and the positive neighbor along each dimension, face_gradient_left is the central cell and the negative neighbor along each dimension

File: pyphasefield/pyphasefield/field.py (hunk 3 above)
Comment (Zhyrek) [reply pyphasefield/pyphasefield/field.py:43]: The specific engine will call either face_gradient_right, face_gradient_left, or cell_gradient, depending on which specific type of finite difference gradient it wants to compute.

**Pull request merged**